tools/src/main/java/org/apache/kafka/trogdor/rest/TaskStateType.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.trogdor.rest;
+
+import java.util.Arrays;
+
+/**
+ * The types of states a single Task can be in
+ */
+public enum TaskStateType {
+    PENDING(Constants.PENDING_VALUE),
+    RUNNING(Constants.RUNNING_VALUE),
+    STOPPING(Constants.STOPPING_VALUE),
+    DONE(Constants.DONE_VALUE);
+
+    TaskStateType(String stateType) {}
+
+    public static class Constants {
+        static final String PENDING_VALUE = "PENDING";
+        static final String RUNNING_VALUE = "RUNNING";
+        static final String STOPPING_VALUE = "STOPPING";
+        static final String DONE_VALUE = "DONE";
+        public static final String[] values = new String[]{
+            PENDING_VALUE, RUNNING_VALUE, STOPPING_VALUE, DONE_VALUE
+        };
+    }
+
+    public static boolean isValid(String taskType) {
```

Comment:
Not needed because you can use `Enum#valueOf` (See https://www.tutorialspoint.com/java/lang/enum_valueof.htm )